src/device_factory.cpp

```
@@ -2,18 +2,22 @@
 #include "zigbee_light.h"
 #include "zigbee_utils.h"
 
+#include <QObject>
+
 namespace DeviceFactory {
 
-QPointer<SmartDevice> create_device(QJsonObject const &data) {
+QPointer<SmartDevice> create_device(QJsonObject const &data, QObject *parent=nullptr) {
```

Cppcheck (2.20.0)

[style] The function 'create_device' is never used.

(unusedFunction)